### PR TITLE
Add array route support

### PR DESCRIPTION
**Pull request by faytekin**
To have more readable codes, small changes have been done. In Route, Controller names are given as arrays.

Example
```php
use App\Api\V1\Controllers\LoginController;
$api->post('login', [LoginController::class, 'login']);
```

### CODE REVIEW

**Comment by taliptako** [comment]
@jasonlew @specialtactics Can you please merge this?

**Comment by specialtactics** [comment]
Sorry, but I don't agree that this is more readable - it creates ambiguity because it is not clear what those array elements mean and what they relate to.

It is also not the way documented on the Laravel docs.

**Comment by faytekin** [comment]
I think there has been a misunderstanding. You can set a route in this way at @specialtactics Laravel's own router. But this feature does not exist in Dingo API. I made it as an option. It does not break the current router. Anyone may use or just continue the current one.

File: src/Routing/Router.php [hunk 1]
```diff
@@ -337,6 +337,11 @@ public function addRoute($methods, $uri, $action)
             $action = ['uses' => $action, 'controller' => $action];
         } elseif ($action instanceof Closure) {
             $action = [$action];
+        } elseif (is_array($action)) {
```
Comment (specialtactics) [inline src/Routing/Router.php]: Can you just add a comment here,

```php
// For this sort of syntax $api->post('login', [LoginController::class, 'login']);
```

**Review by specialtactics — CHANGES_REQUESTED** [review]

**Comment by specialtactics** [comment]
Hey @faytekin that's fine in that case, I just requested one change which is to put a comment donoting what the code does.

Also there is a merge conflict with that file which you need to resolve.

**Review by specialtactics — APPROVED** [review]

**Comment by specialtactics** [comment]
Merged, tagged

**Pull request merged**